rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -144,13 +147,110 @@ void testGetOldFederation(
 
     @Test
     void getOldFederation_previouslySetToNull_returnsNull() {
-        StorageAccessor storageAccessor = new InMemoryStorage();
-        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(null);
         federationStorageProvider.setOldFederation(null);
         Federation oldFederation = federationStorageProvider.getOldFederation(federationConstants, activations);
         assertNull(oldFederation);
     }
 
+    private static Stream<Arguments> providePendingFederationAndFormatArguments() {
+        return Stream.of(
+            Arguments.of(P2SH_ERP_FEDERATION_FORMAT_VERSION, buildMockPendingFederation()),
+            Arguments.of(NON_STANDARD_ERP_FEDERATION_FORMAT_VERSION, buildMockPendingFederation()),
+            Arguments.of(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION, buildMockPendingFederation()),
+            Arguments.of(INVALID_FEDERATION_FORMAT, buildMockPendingFederation()),
+            Arguments.of(EMPTY_FEDERATION_FORMAT, buildMockPendingFederation())
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("providePendingFederationAndFormatArguments")
+    void testGetPendingFederation(
+        int federationFormat,
+        PendingFederation expectedFederation
+    ) {
+
+        // Arrange
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+
+        when(activations.isActive(ConsensusRule.RSKIP123)).thenReturn(false);
+
+        if(federationFormat != INVALID_FEDERATION_FORMAT && federationFormat != EMPTY_FEDERATION_FORMAT) {
+            when(activations.isActive(ConsensusRule.RSKIP123)).thenReturn(true);
+        }
```

Comment:
Not sure I follow this, what's the idea?

Reply:
Updated. Separeated the tests.